ipykernel/tests/test_jsonutil.py

```diff
@@ -80,14 +79,8 @@ def test_encode_images():
     encoded2 = json_clean(encode_images(encoded))
     assert encoded == encoded2
     
-    # test that we don't double-encode base64 str
-    b64_str = {}
-    for key, encoded in encoded.items():
-        b64_str[key] = unicode_to_str(encoded)
-    encoded3 = json_clean(encode_images(b64_str))
-    assert encoded3 == b64_str
```

Comment:
unicode to str is no-op, and then this code becomes strictly identical to the one with encoded2.